nitrocli/src/args.rs

```diff
@@ -129,21 +143,271 @@ impl From<DeviceModel> for nitrokey::Model {
 /// A top-level command for nitrocli.
 #[allow(unused_doc_comments)]
 Command! {Command, [
-  Config => ("config", config),
-  Encrypted => ("encrypted", encrypted),
-  Hidden => ("hidden", hidden),
-  Lock => ("lock", lock),
-  Otp => ("otp", otp),
-  Pin => ("pin", pin),
-  Pws => ("pws", pws),
-  Reset => ("reset", reset),
-  Status => ("status", status),
-  Unencrypted => ("unencrypted", unencrypted),
+  Config(ConfigArgs) => ("config", config),
+  Encrypted(EncryptedArgs) => ("encrypted", encrypted),
+  Hidden(HiddenArgs) => ("hidden", hidden),
+  Lock(LockArgs) => ("lock", lock),
+  Otp(OtpArgs) => ("otp", otp),
+  Pin(PinArgs) => ("pin", pin),
+  Pws(PwsArgs) => ("pws", pws),
+  Reset(ResetArgs) => ("reset", reset),
+  Status(StatusArgs) => ("status", status),
+  Unencrypted(UnencryptedArgs) => ("unencrypted", unencrypted),
 ]}
 
+/// Reads or writes the device configuration
+#[derive(Debug, Default, PartialEq, structopt::StructOpt)]
+pub struct ConfigArgs {
+  #[structopt(subcommand)]
+  subcmd: ConfigCommand,
+}
+
+/// Prints the Nitrokey configuration
+#[derive(Debug, Default, PartialEq, structopt::StructOpt)]
+pub struct ConfigGetArgs {}
+
+/// Changes the Nitrokey configuration
+#[derive(Debug, Default, PartialEq, structopt::StructOpt)]
+pub struct ConfigSetArgs {
+  /// Sets the numlock option to the given HOTP slot
+  #[structopt(short = "n", long)]
+  numlock: Option<u8>,
+  /// Unsets the numlock option
+  #[structopt(short = "N", long, conflicts_with("numlock"))]
+  no_numlock: bool,
+  /// Sets the capslock option to the given HOTP slot
+  #[structopt(short = "c", long)]
+  capslock: Option<u8>,
+  /// Unsets the capslock option
+  #[structopt(short = "C", long, conflicts_with("capslock"))]
+  no_capslock: bool,
+  /// Sets the scrollock option to the given HOTP slot
+  #[structopt(short = "s", long)]
+  scrollock: Option<u8>,
+  /// Unsets the scrollock option
+  #[structopt(short = "S", long, conflicts_with("scrollock"))]
+  no_scrollock: bool,
+  /// Requires the user PIN to generate one-time passwords
+  #[structopt(short = "o", long)]
+  otp_pin: bool,
+  /// Allows one-time password generation without PIN
+  #[structopt(short = "O", long, conflicts_with("otp_pin"))]
+  no_otp_pin: bool,
+}
+
+/// Interacts with the device’s encrypted volume
+#[derive(Debug, Default, PartialEq, structopt::StructOpt)]
+pub struct EncryptedArgs {
+  #[structopt(subcommand)]
+  subcmd: EncryptedCommand,
+}
+
+/// Closes the encrypted volume on a Nitrokey Storage
+#[derive(Debug, Default, PartialEq, structopt::StructOpt)]
+pub struct EncryptedCloseArgs {}
```

Comment:
Would you mind merging in my patch such that you can remove those empty structs? Alternatively just adding it to the list of commits works, too, if it's too much work.

I think one thing lacking is that we do not currently handle meta information in our macros (e.g., doc attributes). I think you can match on
```rust
#[$docs:meta])* 
```
to accomplish that.